### PR TITLE
LCH-6426: Add log record parameter object and make logging more extensible.

### PR DESCRIPTION
**Pull request by saxumVermes**

### CODE REVIEW

File: src/Logging/RequestErrorLogParameter.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<?php
+// phpcs:ignoreFile
```
Comment (narendradesai) [inline src/Logging/RequestErrorLogParameter.php:2]: Any specific reason for ignoring phpcs and not adding doc comments?

Comment (saxumVermes) [reply src/Logging/RequestErrorLogParameter.php:2]: To avoid unnecessary boilerplate.

File: src/ContentHubClientTrait.php [hunk 2]
```diff
@@ -3,6 +3,8 @@
 namespace Acquia\ContentHubClient;
 
 use Acquia\ContentHubClient\Guzzle\Middleware\RequestResponseHandler;
+use Acquia\ContentHubClient\Logging\LoggingHelperTrait;
+use Drupal\Core\Logger\LoggerChannelTrait;
```
Comment (narendradesai) [inline src/ContentHubClientTrait.php]: Unused class

File: src/ContentHubClientTrait.php [hunk 3]
```diff
@@ -3,6 +3,8 @@
...
 use GuzzleHttp\HandlerStack;
 use GuzzleHttp\Promise\PromiseInterface;
```
Comment (narendradesai) [inline src/ContentHubClientTrait.php]: Unused PromiseInterface

File: src/ContentHubClientTrait.php [hunk 4]
```diff
@@ -3,6 +3,8 @@
...
 use GuzzleHttp\Psr7\Response;
```
Comment (narendradesai) [inline src/ContentHubClientTrait.php]: Unused

Comment (narendradesai) [reply src/ContentHubClientTrait.php]: `LogLevel` is also not used

File: src/Logging/LoggingHelperTrait.php [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+   *   Status code.
+   * @param string $reason
+   *   Reason.
+   * @param string|null $request_id
+   *   The request id from the ContentHub service if available.
+   *
+   * @return \GuzzleHttp\Psr7\Response
+   *   Response.
+   */
+  protected function getErrorResponse($code, $reason, $request_id = NULL) {
```
Comment (narendradesai) [inline src/Logging/LoggingHelperTrait.php]: Missing return type

Comment (narendradesai) [reply src/Logging/LoggingHelperTrait.php]: Missing parameter hinting

File: src/Logging/RequestErrorLogParameter.php (hunk 1 above)
Comment (saxumVermes) [reply src/Logging/RequestErrorLogParameter.php:2]: We are using drupal standards here and this file can grow up to twice its size due to that reason, while this is only a parameter object. Drupal standard, in my opinion, has some unnecessary or unrefined restrictions which results in an antipattern of over-documentation.

**Review by narendradesai — CHANGES_REQUESTED** [review]

File: src/ContentHubClientTrait.php (hunk 2 above)
Comment (saxumVermes) [reply src/ContentHubClientTrait.php]: Hmm, we're gonna need to finetune phpcs. These should have been caught.

**Review by narendradesai — APPROVED** [review]
Looks good

**Pull request merged**